chart/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,31 @@
+{{- if .Values.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "docker-hub-rate-limit-exporter.fullname" . }}
+  {{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- end }}
+  labels:
+    {{- include "docker-hub-rate-limit-exporter.labels" . | nindent 4 }}
+    {{- if .Values.serviceMonitor.additionalLabels }}
+    {{- toYaml .Values.serviceMonitor.additionalLabels | nindent 4 }}
+    {{- end }}
+
+spec:
```

Comment:
We are missing a few config options (jobLabel, targetLabel, etc) here, based on the official prometheus [operator documentation](https://github.com/prometheus-operator/prometheus-operator/blob/master/Documentation/api.md#servicemonitorspec). 
I think we should add them, otherwise if somebody needs to configure them, he/she would need to fall back to manually creating the ServiceMonitor resource, right?
Whats your opinion on that @TJM ?

Reply:
@mstein11 We can certainly add these options, but I based my servicemonitor roughly on the output of `kubectl get servicemonitor -o yaml` in the "monitoring" namespace (i.e. the servicemonitor objects that came with kube-prometheus-stack).

I think that `jobLabel` would retrieve the job name (in prometheus) from a label, which might make sense if we were dynamically picking up multiple jobs, but this one is pretty static. With the current configuration, the job name defaults to the same pattern everything else is using (`{namespace}/{servicemonitor name}/0`), or in my case `monitoring/docker-hub-rate-limit-exporter/0` ... which seems good? Also, it is possible that I am misreading the documentation. :)

For `targetLabel`, we are setting the selector with the template, using the same labels we are using for the service itself, so it "should" match?  It seems like they might be redundant options? (but again, maybe I am just misreading)

For what its worth, the "optional,  `additionalLabels` section is there in case you have not set the `prometheus.prometheusSpec.serviceMonitorSelectorNilUsesHelmValues=false` or if you have other selection options in your prometheus settings. That might be useful information to add to the documentation?

Reply:
@TJM thanks, that makes sense. I am not sure about the documentation either but would interpret it the same was as you do. Based on that, I also can not think of a use case where somebody would need to configure those options, so let's just ignore them for now. I will prepare our documentation to reflect the changes. Good point with the additionalLabels section, I'll make sure to add it to the readme. I will merge afterwards. 

Thanks again for contributing :-)